api/controllers/PractitionerController.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * PractitionerController
+ *
+ * @description :: Server-side actions for handling incoming requests.
+ * @help        :: See https://sailsjs.com/docs/concepts/actions
+ */
+
+module.exports = {
+  index: async function (req, res) {
+    let medicalReport = await MedicalReport.findOne({
+      where: {
+        applicationCode: req.session.applicationCode
+      }
+    });
```

Comment:
In the future it would be nice to handle this automatically as a policy or filter so we can handle the lookup and DB errors in one spot

Reply:
Agreed, it'd be nice to move this boiler plate stuff somewhere else... I wasn't sure that a policy was the right place for this type of thing. I suppose we're already querying for the existing MR in the reportExists policy, maybe that policy could just assign it to res.locals? 

Problem being on some controller methods we need to eager load other tables with the MR. Also, I'm a bit leery of having something that a controller method needs just magically available, could make the code less readable.

So yeah, not sure what to do...

Reply:
There's also the flash/merge block a few lines down. I'd love to refactor that out too.

Reply:
Made a refactoring ticket to revisit this.